library/core/src/num/int_macros.rs

```diff
@@ -2477,18 +2477,18 @@ macro_rules! int_impl {
                 None
             } else {
                 let mut n = 0;
-                let mut r = self;
+                let mut r = 1;
 
                 // Optimization for 128 bit wide integers.
                 if Self::BITS == 128 {
                     let b = Self::ilog2(self) / (Self::ilog2(base) + 1);
                     n += b;
-                    r /= base.pow(b as u32);
+                    r *= base.pow(b);
                 }
 
-                while r >= base {
-                    r /= base;
+                while r <= self / base {
```

Comment:
The optimizer should be smart enough to [hoist](https://en.wikipedia.org/wiki/Loop-invariant_code_motion) the division outside of the loop, but as an alternative I can write it explicitly as

```rust
let self_base = self / base;
while r <= self_base {
```